client/state/selectors/is-site-in-profile-links.js

```diff
@@ -0,0 +1,35 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { some } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { getProfileLinks } from 'state/selectors';
+
+/**
+ * Whether the site with the domain in question is currently in the user's profile links.
+ * Will return null if profile links have not been loaded yet.
+ *
+ * @param {Object}   state      Global state tree
+ * @param {String}   siteDomain Site domain
+ * @return {?Boolean}           True if the site is in the user's profile links, false otherwise.
+ */
+export default createSelector(
+	( state, siteDomain ) => {
+		const profileLinks = getProfileLinks( state );
+		if ( profileLinks === null ) {
+			return null;
+		}
+
+		return some( profileLinks, profileLink => {
+			// the regex below is used to strip any leading scheme from the profileLink's URL
+			return siteDomain === profileLink.value.replace( /^.*:\/\//, '' );
```

Comment:
I agree, but I was thinking to leave this for another iteration for the following reasons:

* This selector is a direct port of `UserProfileLinks.prototype.isSiteInProfileLinks`, and it's my intention to not introduce any changes while porting it.
* The API will currently return the schema-less URL `//yourgroovydomain.com` as  `http:////yourgroovydomain.com/` for some reason. While this is probably a bug, it seems that the current regex will catch the schema-less case, too. 
* I agree it's probably too pedantic, because all we want do to here is just obtain the domain out of the URL, which seems to work in all cases outlined.

I'd refrain from any possible improvements on this side, and stick to the purpose of this PR and sprint, which is reduxifing good old `UserProfileLinks`.

What do you think @sirreal?

Reply:
I was curious about how the API handled things, thanks for investigating. That's the place to improve things.

👍  Proceed!